File: LCM And GCD - GFG/lcm-and-gcd.java
//{ Driver Code Starts
import java.io.*;
import java.util.*;

class GFG {
    public static void main(String args[]) throws IOException {
        BufferedReader read =
            new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(read.readLine());
        while (t-- > 0) {
            String S[] = read.readLine().split(" ");
            Long A = Long.parseLong(S[0]);
            Long B = Long.parseLong(S[1]);

            Solution ob = new Solution();
            Long[] ptr = ob.lcmAndGcd(A,B);
            
            System.out.print(ptr[0]);
            System.out.print(" ");
            System.out.println(ptr[1]);
        }
    }
}
// } Driver Code Ends


class Solution {
    static Long[] lcmAndGcd(Long A , Long B) {
        // code here
        
        if(A < B) {
            Long t = A;
            A = B;
            B = t;
        }
        
        Long gcd = Gcd(A,B);
        Long lcm = (A/gcd)*B;
        
        Long[] list = {lcm, gcd};
        return list;
        
        
    }
    
    static Long Gcd(Long a, Long b){
        if(b == 0) return a;
        
        return Gcd(b,a%b);
        
    }
};

File: LCM And GCD - GFG/lcm-and-gcd.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
class Solution {
  public:
  long Getgcd(long long A, long long B){
      if(B == 0) return A;
      return Getgcd(B,A%B);
      
  }
    vector<long long> lcmAndGcd(long long A , long long B) {
        // code here
        
        vector<long long> v(2);
        if(A < B) swap(A,B);
//        cout << A << B;
        long gcd = Getgcd(A,B);
        long lcm = (A/gcd)*B;
        v[0] = lcm;
        v[1] = gcd;
        
        return v;
        
    }
};

//{ Driver Code Starts.
int main() {
    int t;
    cin >> t;
    while (t--) {
        long long A,B;
        
        cin>>A>>B;

        Solution ob;
        vector<long long> ans = ob.lcmAndGcd(A,B);
        cout<<ans[0]<<" "<<ans[1]<<endl;
    }
    return 0;
}
// } Driver Code Ends